mongodb-datastore/handlers/events.go

```diff
@@ -178,6 +173,31 @@ func insertEvent(logger *keptncommon.Logger, event *models.KeptnContextExtendedC
 	return nil
 }
 
+func ensureIndexExistsOnCollection(ctx context.Context, collection *mongo.Collection, indexName string, logger *keptncommon.Logger) {
+	logger.Debug("ensuring index for " + collection.Name() + " exists")
+	indexID := collection.Name() + "-" + indexName
+
+	// if this index has already been created, there is no need to do so again
+	if createdIndexes[indexID] {
+		return
+	}
+
+	indexDefinition := mongo.IndexModel{
+		Keys: bson.M{
+			indexName: 1,
+		},
+	}
+	// CreateOne() is idempotent - this operation checks if the index exists and only creates a new one when not available
+	_, err := collection.Indexes().CreateOne(ctx, indexDefinition)
+	if err != nil {
+		// log the error, but continue anyway - index is not required for the query to work
+		logger.Debug("could not create index for " + collection.Name() + ": " + err.Error())
+	}
```

Comment:
Can we improve this kind of error handling a little bit? Is there a way to check if the error is somethingl ike `already exists`?

If not, does it make sense to rename `createdIndexes` to `skipCreatingIndex`?

Reply:
I just checked - in case the index does already exist, the function does not return an error. renaming `createdIndexes` to `skipCreatingIndex` makes sense, as that will make the purpose of this var easier to understand